docs/REST API/rest-api.md

```diff
@@ -8822,6 +8822,7 @@ Get a list of all software.
 | available_for_install   | boolean | query | If `true` or `1`, only list software that is available for install (added by the user). Default is `false`.                                                                |
 | self_service            | boolean | query | If `true` or `1`, only lists self-service software. Default is `false`.  |
 | packages_only           | boolean | query | If `true` or `1`, only lists packages available for install (without App Store apps).  |
+| platforms               | string | query |  Platform to filter software titles by. Defaults to all platforms.  |
```

Comment:
@rachaelshaw @getvictor We need this API change to [support this UI](https://www.figma.com/design/KqOMIwEQTBAPrUuhmRsS9n/%2323115-Policy-automations%3A-install-App-Store-apps-on-macOS?node-id=5329-2088&node-type=frame&t=8DG55bzK2npr3lPb-11). A bit of context: for policy automation, we're going to support all packages and App Store apps for macOS (excluding iOS and iPadOS). 

So as part of API request, we need to specify multiple platforms. @rachaelshaw Do we have similar behavior in somewhere else?

I was thinking for this one to allow user to specify `platform` multiple times, e.g.:

`/api/v1/fleet/software/titles?team_id=3&platform=darwin&platform=windows&platform=linux`

Reply:
In queries & policies it's a comma-separated string ([example](https://github.com/fleetdm/fleet/blob/d746b9179aed8a60542f8be947b984ad94524e76/it-and-security/lib/collect-fleetd-information.yml#L13))

Reply:
@rachaelshaw It's part of the request body in queries and policies API (POST). I'm just wondering what would be best practice in REST API for these kind of things when it's query.

Something like this?
`/api/v1/fleet/software/titles?team_id=3&platform=darwin,windows,linux`

Reply:
@rachaelshaw Wdyt?

Reply:
The preference from the FE would the first @getvictor suggested (i.e. `?platform=darwin&platform=windows&platform=linux`.)